base/abstractarray.jl

```diff
@@ -392,7 +392,12 @@ julia> stride(A,3)
 12
 ```
 """
-stride(A::AbstractArray, k::Integer) = strides(A)[k]
+function stride(A::AbstractArray, k::Integer)
+    st = strides(A)
+    k < 1 && return 1
```

Comment:
This feels funny to me; what is the meaning of `stride(A, 0)` or `stride(A, -1)`?  I see that this happens to be the behavior that falls out of the current `stride(::DenseArray, n)` implementation, but I'd rather keep the abstract behaviors more conservative and closer to the status quo  — especially to be considered for a back port.

Reply:
I agree, though then we shouldn't call `strides(A::AbstractVector, 2)` in the definition of `strides(A')`.

I could revert back to 
```julia
Base.strides(A::Adjoint{<:Real, <:AbstractVector}) = (1, stride(A.parent, 1))
```
despite it being inconsistent with your `strides(randn(4, 1)')` example (it's multipled by 0 anyways)

Reply:
There are two cases here, `n < 1` and `n > ndims(A)`.  This is only talking about the `n < 1` case — I see the motivation for the `n > ndims(A)` case.

Reply:
Ah ok (I don't see the motivation for either). So perhaps just delete `n < 1` case and let it throw an error?

Reply:
Yes, that'd be my preference. :+1:

Reply:
Bump